cli/cli.go

```diff
@@ -0,0 +1,246 @@
+package cli
+
+import (
+	"os"
+
+	"pkg.re/essentialkaos/ek.v8/arg"
+	"pkg.re/essentialkaos/ek.v8/fmtc"
+	"pkg.re/essentialkaos/ek.v8/usage"
+	"pkg.re/essentialkaos/ek.v8/terminal"
+
+	"github.com/gongled/vgrepo/prefs"
+	"github.com/gongled/vgrepo/repo"
+	//"github.com/gongled/vgrepo/meta"
+)
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+const (
+	APP  = "vgrepo"
+	VER  = "2.0.0"
+	DESC = "Simple CLI tool for managing Vagrant repositories"
+)
+
+const (
+	CMD_ADD    = "add"
+	CMD_DELETE = "delete"
+	CMD_LIST   = "list"
+	CMD_INFO   = "info"
+	CMD_HELP   = "help"
+
+	CMD_ADD_SHORTCUT    = "a"
+	CMD_DELETE_SHORTCUT = "d"
+	CMD_LIST_SHORTCUT   = "l"
+	CMD_INFO_SHORTCUT   = "i"
+)
+
+const (
+	ARG_NO_COLOR = "nc:no-color"
+	ARG_HELP     = "h:help"
+	ARG_VER      = "v:version"
+)
+
+const (
+	ERROR_UNSUPPORTED = 0
+)
+
+const CONFIG_FILE = "vgrepo.knf"
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+var preferences *prefs.Preferences
+
+var argMap = arg.Map{
+	ARG_NO_COLOR: {Type: arg.BOOL},
+	ARG_HELP:     {Type: arg.BOOL, Alias: "u:usage"},
+	ARG_VER:      {Type: arg.BOOL, Alias: "ver"},
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func Init() {
+	args, errs := arg.Parse(argMap)
+
+	if len(errs) != 0 {
+		fmtc.Println("Arguments parsing errors:")
+
+		for _, err := range errs {
+			fmtc.Printf("  %s\n", err.Error())
+		}
+
+		os.Exit(1)
+	}
+
+	if arg.GetB(ARG_NO_COLOR) {
+		fmtc.DisableColors = true
+	}
+
+	if arg.GetB(ARG_VER) {
+		showAbout()
+		return
+	}
+
+	if arg.GetB(ARG_HELP) || len(args) == 0 {
+		showUsage()
+		return
+	}
+
+	switch len(args) {
+	case 0:
+		showUsage()
+		return
+	case 1:
+		processCommand(args[0], nil)
+	default:
+		processCommand(args[0], args[1:])
+	}
+}
+
+// ////////////////////////////////////////////////////////////////////////////////// //
+
+func prepare() {
+	preferences = prefs.New(CONFIG_FILE)
+	errs := preferences.Validate()
+
+	if len(errs) > 0 {
+		for _, err := range errs {
+			terminal.PrintErrorMessage(err.Error())
+		}
+		os.Exit(1)
+	}
+}
+
+// process start source processing
+func processCommand(cmd string, args []string) {
+	var err error
+
+	prepare()
+
+	switch cmd {
+	case CMD_ADD, CMD_ADD_SHORTCUT:
+		err = addCommand(args)
+	case CMD_DELETE, CMD_DELETE_SHORTCUT:
+		err = deleteCommand(args)
+	case CMD_LIST, CMD_LIST_SHORTCUT:
+		err = listCommand()
+	case CMD_INFO, CMD_INFO_SHORTCUT:
+		err = infoCommand(args)
+	case CMD_HELP:
+		showUsage()
+	default:
+		fmtc.Printf("{r}Unknown command{!}\n")
```

Comment:
Use `terminal. PrintErrorMessage` which will print error message to stderr. `fmtc.Printf` output data to `stdout`.